### PR TITLE
Feature/638 refactor applications+support for color.json

### PR DESCRIPTION
**Pull request by inkhey**
Refactor Applications : 
- [X] Content-Type are now always belonging to Application. Those thing are not anymore distinct thing.
- [X] Current_working app are not found in APP_LIST global_var (not nice but actually complicated to not have a global var for this as it's required indirectly for marshmallow_schema validator).
- [X] New ApplicationApi to get App info about App,Content_types and sidebars entry.
- [X] Move model related to app/content-type (not db related) into app_model (outside of model because of a import issue).
- [X] 'require_content_types' now use slug instead of  ContentType objet because it simpler and has content_type are not anymore global_var, it's complicated to link a specifc content_type.
- [X] ALL_CONTENT_TYPES_VALIDATOR Global_var for validator of content-type, who updated after APP_LIST update
- [X] Update tests for sidebar_entry to work dynamically (setting color, path doesn't break test).

Color.json file support related: 
- [X] add color.json to .gitignore
- [X] doc about color.json file
- [X] Add color.json file support in config + param to set custom path
- [X] add color-test.json file for tests
- [X] replace spectra dep with some code (https://github.com/tracim/tracim/issues/776)

### CODE REVIEW

File: backend/tracim_backend/app_models/applications.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    def __init__(
+            self,
+            label: str,
+            slug: str,
+            fa_icon: str,
+            is_active: bool,
+            config: typing.Dict[str, str],
+            main_route: str,
+            app_config: 'CFG',
```
Comment (buxx) [inline backend/tracim_backend/app_models/applications.py]: There is no import of `CFG`. It must be (use if typing.TYPE_CHECKING condition to import it).

File: backend/tracim_backend/app_models/contents.py [hunk 2]
```diff
@@ -209,19 +162,37 @@ class ContentTypeList(object):
...
-    File = file_type
-    Page = html_documents_type
-    Thread = thread_type
 
-    def __init__(self, extend_content_status: typing.List[ContentType]):
-        self._content_types = [self.Folder]
-        self._content_types.extend(extend_content_status)
+    @property
+    def Folder(self):
+        return self.get_one_by_slug('folder')
```
Comment (buxx) [inline backend/tracim_backend/app_models/contents.py]: use `tracim_backend/app_models` variable instead string

File: backend/tracim_backend/app_models/contents.py [hunk 3]
```diff
@@ -134,60 +130,17 @@ def __init__(
...
-    creation_label='Create a folder',
-    available_statuses=CONTENT_STATUS.get_all(),
-    allow_sub_content=True,
-)
-
+thread_type = 'thread'
+file_type = 'file'
+markdownpluspage_type = 'markdownpage'
+html_documents_type = 'html-document'
+folder_type = 'folder'
```
Comment (buxx) [inline backend/tracim_backend/app_models/contents.py]: Prefer uppercase named var (create issue to make it later is you want)

File: backend/tracim_backend/app_models/contents.py [hunk 4]
```diff
@@ -209,19 +162,37 @@ class ContentTypeList(object):
...
+    def __init__(self, app_list: typing.List['Application']):
+        self.app_list = app_list
         self._special_contents_types = [self.Comment]
         self._extra_slugs = [self.Any_SLUG]
 
+    @property
+    def _content_types(self):
+        app_api = ApplicationApi(self.app_list)
+        content_types = app_api.get_content_types()
+        # content_types.extend(self._special_contents_types)
```
Comment (buxx) [inline backend/tracim_backend/app_models/contents.py]: remove comment

File: backend/tracim_backend/extensions.py [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+# TODO - G.M - 2018-08-08 - [GlobalVar] Refactor Global var of tracim_backend
+
+# INFO - G.M - 2018-08-08 - APP_LIST
+# APP_LIST is one of the few "global_val" in tracim_backend, with hapic
+# and ALL_CONTENT_TYPES_VALIDATOR.
+# The goal of this is to be able to get current list of loaded app.
+# List is empty until config load apps.
+# If you need to update APP_LIST, think about updating Content validator like
+# ALL_CONTENT_TYPES_VALIDATOR , see  update_validators() method.
+APP_LIST = []
```
Comment (buxx) [inline backend/tracim_backend/extensions.py]: Should not be in uppercase: uppercase var are constants

File: backend/tracim_backend/app_models/validator.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from marshmallow.validate import OneOf
+from tracim_backend.app_models.contents import CONTENT_TYPES
+
+# TODO - G.M - 2018-08-08 - [GlobalVar] Refactor Global var
+# of tracim_backend, Be careful ALL_CONTENT_TYPES_VALIDATOR is a global_var !
+
+ALL_CONTENT_TYPES_VALIDATOR = OneOf(choices=[])
```
Comment (buxx) [inline backend/tracim_backend/app_models/validator.py]: Should not be in uppercase: uppercase var are constants

File: backend/tracim_backend/lib/utils/utils.py [hunk 7]
```diff
@@ -3,11 +3,11 @@
 import random
 import string
 from enum import Enum
+import colorsys
 
 from redis import Redis
 from rq import Queue
 
-from tracim_backend.config import CFG
```
Comment (buxx) [inline backend/tracim_backend/lib/utils/utils.py]: import with typing.TYPE_CHECKING

File: backend/tracim_backend/app_models/applications.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/app_models/applications.py]: Done

**Pull request merged**